### ISSUE TITLE
How to deal with POST form-data and x-www-form-urlencoded arguments

### ISSUE THREAD

**Issue by rjpcomputing** [kernelsauce/turbo#229]
Hi,

I can't seem to get at the parameters from a POST turbo.web.RequestHandler when they are `form-data` or `x-www-form-urlencoded` parameters.

They come through the "body" but how do I change them into a table. Does Turbo.lua do this for me? I figured it would and I can't seem to figure out how. I found "[turbo.httputil.parse_multipart_data(data)](https://turbo.readthedocs.org/en/latest/httputil.html#parse_multipart_data)" and "[turbo.httputil.parse_post_arguments(data)](https://turbo.readthedocs.org/en/latest/httputil.html#parse_post_arguments)" but that forces me to look at the `Content-Type` header and that seems like it is too deep in the internals for where Turbo.lua wants you to be.

What should I do to create a generic handler for both encoding types?

**Comment by kernelsauce** [comment]
In a RequestHandler you should be able to use e.g self:get_argument.

https://turbo.readthedocs.org/en/latest/web.html#input

**Comment by rjpcomputing** [comment]
I need to get any argument. I am writing an error handler that is generic. So doe "get_arguments()" work?

**Comment by javatmn** [comment]
Try to get the parsed result like below from your POST request handler

local   arg = self.request.connection.arguments

Print out the arg table to get what you need.

Good luck

javatmn

**Comment by rjpcomputing** [comment]
@javatmn 
This works but I think this issue #230 is relevant for this discussion as well.
Is there documentation on this element?

Also, I think that having 1 way, no matter what request method is used, to get the arguments would be really helpful and intuitive. Thoughts?

Thoughts?

**Comment by javatmn** [comment]
Try to get it like below from your post request handler:

local   arg = self.request.connection.arguments


That arguments are already parsed, print it out and you will get what you
want.


From:  rjpcomputing <notifications@github.com>
Reply-To:  kernelsauce/turbo
<reply+00928317f026ec756c93c796cd62c599718c57acee61ddc492cf000000011206fe6b9
2a169ce0648ea01@reply.github.com>
Date:  Wednesday, September 9, 2015 at 3:52 AM
To:  kernelsauce/turbo <turbo@noreply.github.com>
Subject:  Re: [turbo] How to deal with POST form-data and
x-www-form-urlencoded arguments (#229)


I need to get any argument. I am writing an error handler that is generic.
So doe "get_arguments()" work?


Reply to this email directly or view it on GitHub
<https://github.com/kernelsauce/turbo/issues/229#issuecomment-138681790> .

**Comment by javatmn** [comment]
Hi rjpcomputing,

Yes, the document misses second parameter.

I did try to use parse_multipart_data() and parse_post_arguments() by myself
until I found that ³self.request.connection.arguments" already has what I
needed.

I am not Lua expert nor html expert, but I think
³self.request.connection.arguments² will be generally enough(there is code
to decide the boundary during parsing), parse_multipart_data() and
parse_post_arguments() might be used for customisation.

My 2 cents.

javatmn



From:  rjpcomputing <notifications@github.com>
Reply-To:  kernelsauce/turbo
<reply+0092831720b132e0460dadcb083f93aa8c71cb4debbe507292cf000000011208068d9
2a169ce0648ea01@reply.github.com>
Date:  Wednesday, September 9, 2015 at 10:39 PM
To:  kernelsauce/turbo <turbo@noreply.github.com>
Cc:  "'javatmn@gmail. com'" <javatmn@gmail.com>
Subject:  Re: [turbo] How to deal with POST form-data and
x-www-form-urlencoded arguments (#229)


@javatmn <https://github.com/javatmn>
This works but I think this issue #230
<https://github.com/kernelsauce/turbo/issues/230>  is relevant for this
discussion as well.
Is there documentation on this element?

Also, I think that having 1 way, no matter what request method is used, to
get the arguments would be really helpful and intuitive. Thoughts?

Thoughts?


Reply to this email directly or view it on GitHub
<https://github.com/kernelsauce/turbo/issues/229#issuecomment-138931214> .
